src/processors/default/processPlatformTracks/processPlatformTracks.ts

```diff
@@ -62,11 +63,12 @@ const processorFunction = (platform: MusicPlatform) => async (nfts: NFT[], clien
     let nftFactoryType: MusicPlatformTypeConfig;
     try {
       nftFactoryType = getNFTFactoryType(nftFactory, platformType);
+      const extractor = idExtractor(nftFactory);
```

Comment:
i think it's safe to assume that all nfts within a single nftFactory will use the same idExtractor.

given that, we could just send the nftFactory into getTrackInputs and into the mapNFTsToTrackIds function and let it handle getting the idExtractor